sql/moz-fx-data-shared-prod/firefox_accounts_derived/fxa_users_services_daily_v1/query.sql

```diff
@@ -68,50 +74,166 @@ WITH base AS (
+  USING
+    (flow_id)
+  GROUP BY
+    user_id,
+    service
+),
+windowed AS (
+  SELECT
+    `timestamp`,
+    user_id,
+    `service`,
     udf.mode_last(ARRAY_AGG(country) OVER w1) AS country,
-    udf.mode_last(ARRAY_AGG(LANGUAGE) OVER w1) AS language,
+    udf.mode_last(ARRAY_AGG(`language`) OVER w1) AS `language`,
     udf.mode_last(ARRAY_AGG(app_version) OVER w1) AS app_version,
     udf.mode_last(ARRAY_AGG(os_name) OVER w1) AS os_name,
     udf.mode_last(ARRAY_AGG(os_version) OVER w1) AS os_version,
+    udf.mode_last(ARRAY_AGG(ua_version) OVER w1) AS ua_version,
+    udf.mode_last(ARRAY_AGG(ua_browser) OVER w1) AS ua_browser,
     udf_contains_tier1_country(ARRAY_AGG(country) OVER w1) AS seen_in_tier1_country,
-    udf_contains_registration(ARRAY_AGG(event_type) OVER w1) AS registered
+    LOGICAL_OR(event_type = 'fxa_reg - complete') OVER w1 AS registered,
+    ARRAY_AGG(event_type) OVER w1 AS service_events,
   FROM
-    base
+    fxa_events
   WHERE
-    service IS NOT NULL
-    -- Reprocess all dates by running this query with --parameter=submission_date:DATE:NULL
-    AND (@submission_date IS NULL OR @submission_date = DATE(`timestamp`))
+    DATE(`timestamp`) = @submission_date
+    AND user_id IS NOT NULL
+    AND `service` IS NOT NULL
+  QUALIFY
+    ROW_NUMBER() OVER (PARTITION BY user_id, `service`, DATE(`timestamp`) ORDER BY `timestamp`) = 1
   WINDOW
     w1 AS (
       PARTITION BY
         user_id,
-        service,
+        `service`,
         DATE(`timestamp`)
       ORDER BY
-        `timestamp` --
+        `timestamp`
       ROWS BETWEEN
         UNBOUNDED PRECEDING
         AND UNBOUNDED FOLLOWING
-    ),
-    -- We must provide a modified window for ROW_NUMBER which cannot accept a frame clause.
-    w1_unframed AS (
-      PARTITION BY
-        user_id,
-        service,
-        DATE(`timestamp`)
-      ORDER BY
-        `timestamp`
     )
 )
 SELECT
-  * EXCEPT (_n)
+  DATE(@submission_date) AS submission_date,
+  windowed.user_id,
+  windowed.`service`,
+  windowed.`timestamp`,
+  windowed.country,
+  windowed.`language`,
+  windowed.app_version,
+  windowed.os_name,
+  windowed.os_version,
+  windowed.ua_version,
+  windowed.ua_browser,
+  windowed.seen_in_tier1_country,
+  windowed.registered,
+  -- needed for first_seen logic
+  windowed.service_events,
+  -- info about first user/service flow observed
+  -- on a specific day. Needed for first_seen logic
+  STRUCT(
+  -- flow entrypoints
+    user_service_flow_entrypoints.flow_entrypoint_info.flow_id,
+    user_service_flow_entrypoints.flow_entrypoint_info.`timestamp` AS flow_timestamp,
+    user_service_flow_entrypoints.flow_entrypoint_info.entrypoint,
+    -- flow utms
+    user_service_flow_utms.utm_info.utm_term,
+    user_service_flow_utms.utm_info.utm_medium,
+    user_service_flow_utms.utm_info.utm_source,
+    user_service_flow_utms.utm_info.utm_campaign,
+    user_service_flow_utms.utm_info.utm_content
+  ) AS first_daily_service_flow_info,
```

Comment:
I think I can see what you mean now. For some reason I thought I was joining the utm info using flow_id rather than user, service combination for joining the data.

I guess the question would be here if that's what I should be doing?